projects/uvis_gs_fail/training_validation.py

```diff
@@ -0,0 +1,613 @@
+            
+            # Calculate output and loss from validation batch
+            output = model(data)
+            val_loss += distance(output, target).item()
+            
+            # Count number of correct predictions
+            pred = output.data.max(1, keepdim=True)[1]
+            correct += pred.eq(target.data.view_as(pred)).sum()
+        
+        # Calculate accuracy
+        accuracy = 100. * correct / len(valid_loader.dataset)
+        
+        # Calculate validation and recall for the epoch
+        prec_global,rec_global,fscore,support = precision_recall_fscore_support\
+                             (target,pred, average = 'micro', zero_division=1.0)
+        
+        precision = prec_global
+        recall = rec_global
+        fscores = fscore
+
+    # Normalize validation loss from one epoch
+    val_loss_norm = val_loss / len(valid_loader)
+    
+    return val_loss_norm, accuracy, precision, recall, fscores
+
+def main(path, metric_dir, data_type = 'None'):
+    '''Load, organize, and format data sets, and train a model on that data.
+
+    Performs the following steps for a model:
+    1. Calls organize_data to get all the processed data and puts them into 
+    tuples for augmented and non-augmented data.
+    2. Uses a boolean to decide wether to use augmented or non-augmented data, 
+    and splits the datasets into x and y datasets for training, and validation 
+    sets.
+    3. Set the hyperparameters, and format the training and validation sets as
+    callable methods.
+    4. Run the training and validation loops for the specified number of epochs
+    and save the state dictionary every 10 epochs.
+    5. Save the training and validation evaluation metrics for each epoch of 
+    training to csv files.
+
+    Parameters
+    -----------
+    path : list
+        Identifies the directory that the csv file with the path to each image
+        is located in.
+    metric_dir : string
+        The directory to save model evaluation metrics, data indices, and state
+        dictionaries to.
+    data_type : string
+        The type of data that is being used to train the model. Default='None'
+
+    Returns
+    --------
+    metric_data : Dataframe
+        The data for the loss, accuracy, precision, and recall metrics for both 
+        the training and validation loops at each epoch of training.
+    '''
+    # Main directory to read files from
+    augmented_data, non_augmented_data = organize_data(path)
+    
+    if data_type == 'aug':
+        x_train = augmented_data[0]
+        y_train = augmented_data[1]
+        
+        x_test = augmented_data[2]
+        y_test = augmented_data[3]
+        
+    elif data_type == 'non_aug':
+        x_train = non_augmented_data[0]
+        y_train = non_augmented_data[1]
+        
+        x_test = non_augmented_data[2]
+        y_test = non_augmented_data[3]
+            
+    elif data_type == 'None':
```

Comment:
Remove this elif and statement since `data_type` error is getting caught in `main`.